src/single.c

```diff
@@ -0,0 +1,87 @@
+/* SPDX-License-Identifier: CC0-1.0 */
+#include "types.h"
+#include "render.h"
+#include "asset.h"
+#include <string.h>
+
+int mmdoc_single(char *out, char *toc_path, Array toc_refs,
+                 AnchorLocationArray anchor_locations) {
+  char asset_path[2048];
+  FILE *asset_file;
+  strcpy(asset_path, out);
+  strcat(asset_path, "/highlight.pack.js");
+  asset_file = fopen(asset_path, "w");
```

Comment:
At some point you should also check the return value of all fopen/fputc calls. They can fail.